src/read/dwarf.rs

```diff
@@ -87,14 +89,25 @@ impl<T> Dwarf<T> {
             debug_line_str: Section::load(&mut section)?,
             debug_str: Section::load(&mut section)?,
             debug_str_offsets: Section::load(&mut section)?,
-            debug_str_sup: Section::load(&mut sup)?,
             debug_types: Section::load(&mut section)?,
             locations: LocationLists::new(debug_loc, debug_loclists),
             ranges: RangeLists::new(debug_ranges, debug_rnglists),
             file_type: DwarfFileType::Main,
+            sup: None,
         })
     }
 
+    /// Load the DWARF sections from the supplementary object file.
+    ///
+    /// `section` operates the same as for `load`.
+    pub fn load_sup<F, E>(&mut self, section: F) -> core::result::Result<(), E>
+    where
+        F: FnMut(SectionId) -> core::result::Result<T, E>,
+    {
+        self.sup = Some(Arc::new(Self::load(section)?));
```

Comment:
Should this assert that `self.sup.is_none()`? Or document that it will overwrite any already-loaded supplementary info? Not sure what the right choice is here. (But if we assert, that should also be documented.)

Reply:
Sure, documented it.